src/sys/object.rs

```diff
@@ -10,6 +10,18 @@ pub unsafe fn new(out: &mut Local, env: Env) {
     napi::create_object(env, out as *mut _);
 }
 
+#[cfg(feature = "napi-8")]
+pub unsafe fn freeze(env: Env, obj: Local) -> bool {
+    let status = napi::object_freeze(env, obj);
+    status == napi::Status::Ok
```

Comment:
I think `GenericFailure` and `ExceptionPending` are the only valid responses. Everything else would be an error on Neon's part.

Maybe we could return `Status` and let the higher level handle that match?

Reply:
I agree this would be a good change.